### PR TITLE
Intruduce new processor for insert interpreter(base on pr4862)

### PR DESCRIPTION
**Pull request by xudong963**

### CODE REVIEW

File: query/src/pipelines/new/processors/sources/sync_ck_source_receiver.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        })
+    }
+}
+
+#[async_trait::async_trait]
+impl SyncSource for SyncReceiverCkSource {
+    const NAME: &'static str = "SyncReceiverCkSource";
+
+    fn generate(&mut self) -> Result<Option<DataBlock>> {
+        match self.receiver.blocking_recv() {
```
Comment (sundy-li) [inline query/src/pipelines/new/processors/sources/sync_ck_source_receiver.rs:54]: Good!~

File: query/src/servers/clickhouse/interactive_worker_base.rs [hunk 2]
```diff
@@ -92,22 +98,81 @@ impl InteractiveWorkerBase {
...
+                None => panic!("Interpreter isn't a InsertInterpreter!"),
+            };
+            let mut insert_interpreter_box = insert_interpreter.clone().get_box();
+            // Set `SourcePipeBuilder` to `InsertInterpreter`, used in insert source is `StreamingWithFormat`
+            insert_interpreter_box
+                .as_mut()
+                .set_source_pipe_builder(Some(source_pipe_builder));
+
+            // Set the newest `InsertInterpreter` to `InterceptorInterpreter`
+            let mut interceptor_interpreter_box = interceptor_interpreter.clone().get_box();
```
Comment (sundy-li) [inline query/src/servers/clickhouse/interactive_worker_base.rs:147]: This is too hacky. 

I think we can just add one method and make it simple.

```
pub fn set_source_pipe_builder(&self, builder: Option<SourcePipeBuilder>) -> Self {
        let mut res = self.clone();
        res.source_pipe_builder = builder; 
        res
    }
```

File: query/src/interpreters/interpreter_insert.rs [hunk 3]
```diff
@@ -83,10 +99,25 @@ impl InsertInterpreter {
...
+                        .get_box()
+                        .as_mut()
+                        .source_pipe_builder
+                        .take()
+                        .ok_or_else(|| ErrorCode::EmptyData("empty source pipe builder"))?
+                        .finalize(),
+                );
+            }
+            InsertInputSource::SelectPlan(plan) => {
+                let builder = QueryPipelineBuilder::create(self.ctx.clone());
```
Comment (sundy-li) [inline query/src/interpreters/interpreter_insert.rs:114]: duplicate codes with `SelectInterpreter`.

We can create `SelectInterpreter` and pub the method `create_new_pipeline`.

File: query/src/servers/clickhouse/interactive_worker_base.rs (hunk 2 above)
Comment (xudong963) [reply query/src/servers/clickhouse/interactive_worker_base.rs:147]: So we need to add `set_source_pipe_builder` in `interpreter` trait?

Comment (sundy-li) [reply query/src/servers/clickhouse/interactive_worker_base.rs:147]: Yes,or we can make it:

default function:
```
pub fn set_source_pipe_builder(&self, builder: Option<SourcePipeBuilder>) {        
    }
```


insert/interpreter executor override:

```
pub fn set_source_pipe_builder(&self, builder: Option<SourcePipeBuilder>) {    
	    let mut lock = self. source_pipe_builder.write();  // RWMutext
       ....

    }
```

Comment (xudong963) [reply query/src/servers/clickhouse/interactive_worker_base.rs]: Nice, will use the second way to make code simple!

**Pull request merged**